libs/grok/src/main/java/org/elasticsearch/grok/MatcherWatchdog.java

```diff
@@ -94,58 +76,35 @@ public long maxExecutionTimeInMillis() {
         public void unregister(Matcher matcher) {}
     }
 
-    class Default implements MatcherWatchdog {
+    final class Default implements MatcherWatchdog {
+
+        // duplicated from org.elasticsearch.core.TimeValue because we don't have access to that here
+        private static final long NSEC_PER_MSEC = TimeUnit.NANOSECONDS.convert(1, TimeUnit.MILLISECONDS);
 
-        private final long interval;
         private final long maxExecutionTime;
-        private final LongSupplier relativeTimeSupplier;
-        private final BiConsumer<Long, Runnable> scheduler;
-        private final AtomicInteger registered = new AtomicInteger(0);
-        private final AtomicBoolean running = new AtomicBoolean(false);
-        final ConcurrentHashMap<Matcher, Long> registry = new ConcurrentHashMap<>();
-
-        private Default(long interval, long maxExecutionTime, LongSupplier relativeTimeSupplier, BiConsumer<Long, Runnable> scheduler) {
-            this.interval = interval;
+
+        private Default(long maxExecutionTime) {
```

Comment:
Same here about renaming to include units.

Reply:
Also covered in 19cc20748af329db9248e091e02868ed6094d336.